jwst/tweakreg/tweakreg_step.py

```diff
@@ -70,18 +71,20 @@ def process(self, input):
             # entries when aligning to GAIA
             self.expand_refcat = True
 
+        if len(images) == 0:
+            raise ValueError("Input must contain at least one image model.")
+
         # Build the catalogs for input images
         for image_model in images:
             catalog = make_tweakreg_catalog(
                 image_model, self.kernel_fwhm, self.snr_threshold,
                 brightest=self.brightest, peakmax=self.peakmax
             )
 
-            # filter out sources outside the image array if WCS validity
-            # region is provided:
-            wcs_bounds = image_model.meta.wcs.pixel_bounds
-            if wcs_bounds is not None:
-                ((xmin, xmax), (ymin, ymax)) = wcs_bounds
+            # filter out sources outside the WCS bounding box
+            bb = image_model.meta.wcs.bounding_box
```

Comment:
Good catch!

Reply:
By accident!  It doesn't seem to have changed any of the results in our regression tests, and I thought it might for MIRI, but I think we are filtering any sources from areas outside the bounding box via the very high threshold we use in the regression test.

https://github.com/spacetelescope/jwst/blob/b5fe85dde47bcc6d122464a2ab389bb26b5eae5a/jwst/regtest/test_miri_image.py#L66-L69

It might be good to check to see if it solves the reason that we had to set those high thresholds in the first place.

Reply:
Answering my own question here, the MIRI wcs is defined across the whole image, including the upper left coronagraphic mask, so bounding box has no effect.  The NON_SCIENCE DQ mask is the only thing that saves us in the final output image.  I suspect that we'll have to wait until we get to be on sky to figure out if this is sufficient, or use a special bounding box just for MIRI here in `tweakreg`.